File: frontend/src/services/leadService.js
import api from './api';

const leadService = {
  createLead: async (leadData) => {
    return await api.post('/api/v1/leads', leadData);
  },

  updateLead: async (id, leadData) => {
    return await api.put(`/api/v1/leads/${id}`, leadData);
  },

  getLeadById: async (id) => {
    return await api.get(`/api/v1/leads/${id}`);
  },

  searchLeads: async (params = {}) => {
    return await api.get('/api/v1/leads', { params });
  },

  updateLeadStatus: async (id, status) => {
    return await api.put(`/api/v1/leads/${id}/status`, null, { params: { status } });
  },

  deleteLead: async (id) => {
    return await api.delete(`/api/v1/leads/${id}`);
  },

  restoreLead: async (id) => {
    return await api.put(`/api/v1/leads/${id}/restore`);
  },

  assignLead: async (id, assignedSalesRepId) => {
    return await api.put(`/api/v1/leads/${id}/assign`, { assignedSalesRepId });
  },

  convertLead: async (id, conversionData) => {
    return await api.post(`/api/v1/leads/${id}/convert`, conversionData);
  },
};

export default leadService;

File: src/main/java/com/enterprise/crm/v1/lead/service/LeadService.java
package com.enterprise.crm.v1.lead.service;

import com.enterprise.crm.v1.lead.dto.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.time.LocalDateTime;
import java.util.UUID;

public interface LeadService {
    LeadResponse createLead(CreateLeadRequest request);
    LeadResponse updateLead(UUID id, CreateLeadRequest request);
    LeadResponse getLeadById(UUID id);
    Page<LeadResponse> searchLeads(
            String companyName,
            String status,
            UUID assignedSalesRepId,
            String tag,
            LocalDateTime startCreated,
            LocalDateTime endCreated,
            LocalDateTime startUpdated,
            LocalDateTime endUpdated,
            String source,
            String email,
            Pageable pageable
    );
    void updateLeadStatus(UUID id, String status);
    void deleteLead(UUID id);
    void restoreLead(UUID id);
    void assignLead(UUID id, UUID assignedSalesRepId);
}
